### PR TITLE
Allow more classes (1000 -> 2^20), but won't show the CM if too many. HEXDEV-658

### PR DESCRIPTION
**Pull request by arnocandel**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/521)

### CODE REVIEW

**Comment by arnocandel** [comment]
@h2o-ops please test

File: h2o-core/src/main/java/hex/ConfusionMatrix.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
   private TwoDimTable _table;
   public final double[][] _cm; // [actual][predicted], typed as double because of observation weights (which can be doubles)
   public final String[] _domain;
+  public static final int MAX_CM_CLASSES = 1000;
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/ConfusionMatrix.java]: Does it make sense to make the value configurable via `-D` property?

File: h2o-core/src/main/java/hex/ConfusionMatrix.java [hunk 2]
```diff
@@ -54,15 +56,20 @@ public static ConfusionMatrix buildCM(Vec actuals, Vec predictions) {
         if( !ca.isNA(i) )
           _arr[(int)ca.at8(i)][(int)cp.at8(i)]++;
     }
-    @Override public void reduce( CMBuilder cm ) { ArrayUtils.add(_arr,cm._arr); }
+    @Override public void reduce( CMBuilder cm ) {
+      if (_arr != null && cm._arr != null) ArrayUtils.add(_arr,cm._arr);
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/ConfusionMatrix.java]: Minor: It would be better to modify `ArrayUtils` to deal with `null` inputs.

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Pull request merged**